File: Tools/DevCg/myCg/clsListing.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Text;
using ICg;
namespace myCg
{
    
    public class clsListing : BasicFrame, Frame
    {
        public override void GenCreate_Beforelines()
        {
            outText.AppendLine("TableName = \"" + tablename +   "\";");
        }
        public override void GenCreate_loop_Fistline()
        {
            if (type.ToLower() == "nodb")
            {
                return;
            }
            outText.AppendFormat("DataTableColumns.Add( \"" + "{0}" + "\",{1}," + "\"{2}" + "\",true ,true);", dbName, crudeType, colName);
            outText.AppendLine();
        }
        public override void GenCreate_loop()
        {
            if (type.ToLower() == "nodb")
            {
                return;
            }
            outText.AppendFormat("DataTableColumns.Add( \"" + "{0}\"" + ",{1},\"" + "{2}\""   + ");", dbName, crudeType, colName);
            outText.AppendLine();
        }
        public override void GenCreate_loop_Lastline()
        {
            GenCreate_loop();
        }
        public override void GenCreate_Afterlines()
        {

        }
    }
}

File: cgPr/cgv5fin/CGProject/Interfaced/clsListing.vb
Imports System.Text
Public Class clsListing
    Inherits BasicFrame
    Implements Frame
    Public Overrides Sub GenCreate_Beforelines()
        outText.AppendLine("TableName = " & q & tablename & q & ";")
    End Sub
    Public Overrides Sub GenCreate_loop_Fistline()
        outText.AppendFormat("DataTableColumns.Add( " & q & "{0}" & q & ",{1}," & q & "{2}" & q & ",true ,true);", dbname, CrudeType, colName)
        outText.AppendLine()
    End Sub
    Public Overrides Sub GenCreate_loop()
        outText.AppendFormat("DataTableColumns.Add( " & q & "{0}" & q & ",{1}," & q & "{2}" & q & ");", dbname, CrudeType, colName)
        outText.AppendLine()
    End Sub
    Public Overrides Sub GenCreate_loop_Lastline()
        GenCreate_loop()
    End Sub
    Public Overrides Sub GenCreate_Afterlines()
       
    End Sub
End Class
